### PR TITLE
Generalized config and bug fix

### PR DESCRIPTION
**Pull request by roobyz**
### Idea: Treat "Installation files" like Cattle
Enhance the _config.yaml_ file to centralize the initial configuration settings onto one file and avoid having to manually update the installation files (footloose.yaml, machines.yaml). Added new _parse_yaml()_ function to funtions.sh for processing the config.yaml. Corresponding updates to `setup.sh`, and `setup.js` were made to process the enhanced _config.yaml_.

Benefit that centralized configuration settings are leveraged by `jk` to systematically generate the _footloose.yaml_ and _machine.yaml_ files. And, `rm machines.yaml footloose.yaml` added to `cleanup.sh` to ensure that stale data doesn't hang around.

### Bug "fix" workaround
Due to failure of `wksctl init` step, added two new function to `functions.sh` (_set_flux_version_, _set_wksctl_version_) which get called in `setup.sh` in lieu of `wksctl init` to update:
- **_flux.yaml_**: memcached image version, and flux image version and repo url
- **_wks-controller.yaml_**: wksctl image version

Also, updated the release versions to latest supported versions since the older releases where no longer supported.

### Notes
Added a `notes.md` file to highlight key ideas, prerequisites, and context behind how this all works.

### CODE REVIEW

File: notes.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
+# Firekube
+
+## Files
+
+1. Config.yaml:
+   1. backend type (ignite or docker)
+   2. quantiy and details of nodes allocated for the Kubernetes control plane and workers
```
Comment (paulcarlton-ww) [inline notes.md:7]: typo, quantity spelling

File: notes.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+## Install Helm
+
+``` bash
+helm repo add stable https://kubernetes-charts.storage.googleapis.com/
+helm repo update              # Make sure we get the latest list of charts
+helm search repo stable | grep -v "DEPRECATED"
+```
+
+In stall weave scope:
```
Comment (paulcarlton-ww) [inline notes.md:103]: typo: Install

File: wks-controller.yaml [hunk 3]
```diff
@@ -42,7 +42,7 @@ spec:
         operator: Exists
       containers:
       - name: controller
-        image: quay.io/wksctl/controller:latest
+        image: docker.io/weaveworks/wksctl-controller:v0.8.2-beta.7
```
Comment (paulcarlton-ww) [inline wks-controller.yaml:45]: maybe use the value from config.yaml

**Comment by paulcarlton-ww** [comment]
Some good ideas here but we should fix wksctl bug rather than implement a workaround

File: notes.md (hunk 2 above)
Comment (roobyz) [reply notes.md]: Thanks... corrected see commit: b8f7235

File: notes.md (hunk 1 above)
Comment (roobyz) [reply notes.md]: Thanks... corrected see commit: b8f7235

File: wks-controller.yaml (hunk 3 above)
Comment (roobyz) [reply wks-controller.yaml:45]: Yes... that is what it does... :-) See line 137 of setup.sh:
`set_wksctl_version ${config_images_wksctl}`

The `parse_yaml()` function, reads the whole config.yaml and generates a variable for each value including **_config_images_wksctl_** using the location in the yaml hierarchy to create the variable name. (e.g. "config_" + key1 + key2 + key3...) and then assigns the value.

**Pull request closed without merging**